pkg/export/export.go

```diff
@@ -335,17 +323,25 @@ func (e *Exporter) SetLabelsByIDFunc(f func(uint64) labels.Labels) {
 	e.seriesCache.getLabelsByRef = f
 }
 
-func (e *Exporter) getExternalLabels() labels.Labels {
-	e.mtx.Lock()
-	defer e.mtx.Unlock()
-	return e.externalLabels
-}
-
 // Export enqueues the samples to be written to Cloud Monitoring.
 func (e *Exporter) Export(metadata MetadataFunc, batch []record.RefSample) {
 	if e.opts.Disable {
 		return
 	}
+
+	e.mtx.Lock()
+	externalLabels := e.externalLabels
+	start, end, ok := e.opts.HighAvailabilityRange.Range()
+	e.mtx.Unlock()
+
+	if !ok {
+		// This will clear repeatedly while we don't hold the lease, which doesn't
+		// harm in any way.
+		e.seriesCache.clear()
```

Comment:
So that the start timestamps for cumulatives are purged and we don't send with old ones if become leader again later. Made the comment more descriptive.